test/json-decode.test.ts

```diff
@@ -345,6 +346,74 @@ describe('array', () => {
   });
 });
 
+describe('tuple', () => {
+  describe('when given valid JSON', () => {
+    it('can decode a simple tuple', () => {
+      const decoder = tuple([number(), number()]);
```

Comment:
Thank you for the tests, this is great.
change request: for these tests can you annotate the decoder type? For example
```
const decoder: Decoder<[number, number]> = tuple([number(), number()]);
```
We don't have type-level tests right now, so making sure that the decoders type as expected has value.